packages/circuits/src/initial.circom

```diff
@@ -0,0 +1,89 @@
+pragma circom 2.0.0;
+
+include "./merkle.circom";
+include "../node_modules/circomlib/circuits/comparators.circom";
+
+template CheckReplaceArray(n) {
+    signal input index;
+    signal input newValue;
+    signal input array[n];
+    signal output newArray[n];
+    signal output replacedValue;
+
+    component checkIndex[n];
+    signal items[n];
+    for (var i = 0; i < n; i++) {
+        checkIndex[i] = IsEqual();
+        checkIndex[i].in[0] <== i;
+        checkIndex[i].in[1] <== index;
+        newArray[i] <== (newValue - array[i]) * checkIndex[i].out + array[i];
+        // newArray[i] <== (checkIndex[i].out * newValue) + ((1 - checkIndex[i].out) * array[i]);
+        items[i] <== checkIndex[i].out * array[i];
+    }
+
+    // get the replaced value
+    var temp = 0;
+    for (var i = 0; i < n; i++) {
+        temp += items[i];
+    }
+    replacedValue <== temp;
+}
+
+template Initial(levels, cardCount) {
+    signal input deckRoot;
+    signal input newDeckRoot;
+    signal input deckLeaves[2**levels];
+    signal input initialDeckTailCardIndex;
+    signal input handRoot;
+    signal input newHandRoot;
+    signal input drawnCardIndices[cardCount];
```

Comment:
This is missing the randomness as an input, and in general the verification that the indices were selected correctly.

However, this can probably be added in another component that then includes this component!